surf/apps/communities/views.py

```diff
@@ -161,21 +155,54 @@ def _delete_collections(instance, collections):
         collections = Collection.objects.filter(id__in=collections).all()
         instance.collections.remove(*collections)
 
-    @staticmethod
-    def _check_access(user, instance=None, collection_ids=None):
-        """
-        Check if user is active and admin of community
-        :param user: user
-        :param instance: community instance
-        :param collection_ids: list of identifiers of collections
-        added/deleted to/from community
-        """
-        if not user or not user.is_authenticated:
-            raise AuthenticationFailed()
+
+class CommunityDetailAPIView(APIView):
```

Comment:
The APIView is as bare metal as it gets when it comes to DRF. That's very convenient when dealing with APIViews that spit out predictions from a model. However when dealing with CRUD there's a lot of convenience in [generic API views](https://www.django-rest-framework.org/api-guide/generic-views/). With a few lines you can declare how you want your API to update instead of defining how updates should work.

A RetrieveUpdateAPIView would be the better candidate, but perhaps this entire view can disappear if we use nested updates (see comments above).